fretboard/src/main/java/mozilla/components/service/fretboard/JSONExperimentParser.kt

```diff
@@ -0,0 +1,103 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.fretboard
+
+import org.json.JSONArray
+import org.json.JSONObject
+
+/**
+ * Default JSON parsing implementation
+ */
+class JSONExperimentParser {
+    /**
+     * Creates an experiment from its json representation
+     *
+     * @param jsonObject experiment json object
+     * @return created experiment
+     */
+    fun fromJson(jsonObject: JSONObject): Experiment {
+        val bucketsObject: JSONObject? = jsonObject.optJSONObject("buckets")
+        val matchObject: JSONObject? = jsonObject.optJSONObject("match")
+        val regions: List<String>? = matchObject?.optJSONArray("regions")?.toList()
+        val matcher = if (matchObject != null) {
+            Experiment.Matcher(matchObject.tryGetString("lang"), matchObject.tryGetString("appId"), regions)
```

Comment:
Let's move all those strings ("lang", "appId", ..) into constants in a companion object. Either here or into another class.